lib/pgBackRest/Storage/Posix/Driver.pm

```
@@ -0,0 +1,864 @@
+        $strFile,
+        $bIgnoreMissing,
+    ) =
+        logDebugParam
+        (
+            __PACKAGE__ . '->remove', \@_,
+            {name => 'strFile', trace => true},
+            {name => 'bIgnoreMissing', optional => true, default => false, trace => true},
+        );
+
+    # Working variables
+    my $bRemoved = true;
+
+    # Remove the file
+    if (unlink($strFile) != 1)
+    {
+        $bRemoved = false;
+
+        # If path exists then throw the error
+        if (!($OS_ERROR{ENOENT} && $bIgnoreMissing))
+        {
+            logErrorResult($OS_ERROR{ENOENT} ? ERROR_FILE_MISSING : ERROR_FILE_OPEN, "unable to remove '${strFile}'", $OS_ERROR);
+        }
+    }
+
+    # Return from function and log return values if any
+    return logDebugReturn
+    (
+        $strOperation,
+        {name => 'bRemoved', value => $bRemoved, trace => true}
+    );
+}
+
+####################################################################################################################################
+# list - list a directory
+####################################################################################################################################
+sub list
+{
+    my $self = shift;
+
+    # Assign function parameters, defaults, and log debug info
+    my
+    (
+        $strOperation,
+        $strPath,
+        $bIgnoreMissing,
+    ) =
+        logDebugParam
+        (
+            __PACKAGE__ . '->list', \@_,
+            {name => 'strPath', trace => true},
+            {name => 'bIgnoreMissing', optional => true, default => false, trace => true},
+        );
+
+    # Working variables
+    my @stryFileList;
+    my $hPath;
+
+    # Attempt to open the path
+    if (opendir($hPath, $strPath))
+    {
+        @stryFileList = grep(!/^(\.)|(\.\.)$/i, readdir($hPath));
+        close($hPath);
+    }
+    # Else process errors
+    else
+    {
+        # Ignore the error is the file is missing and missing files should be ignored
```

Comment should be "Ignore the error if the" not "is"